modules/imgcodecs/src/loadsave.cpp

```diff
@@ -452,8 +452,6 @@ static bool imwrite_( const String& filename, const Mat& image,
     Mat temp;
     const Mat* pimage = &image;
 
-    CV_Assert( image.channels() == 1 || image.channels() == 3 || image.channels() == 4 );
```

Comment:
OpenCV includes many I/O backends, not PNG only. Their implementations assume this condition, so they may crash instead of "badarg" assertion.
Need to check all of them how they handle "unexpected" formats.